addons/web/static/src/js/views/list/list_controller.js

```diff
@@ -99,6 +99,33 @@ var ListController = BasicController.extend({
             return record.res_id;
         });
     },
+
+    /**
+     * @typedef {Object} SelectedIds
+     * @property {integer[]} resIds
+     * @property {integer} total - The total number of selected records
+     * @property {boolean} displayNotif - True if not all the available records have been returned
+     */
+
+    /**
+     * Returns the list of currently selected res_ids (with the check boxes on
+     * the left) or the whole domain res_ids if it is selected
+     *
+     * @returns {Promise<SelectedIds>}
+     */
+    getSelectedIdsWithDomain: async function () {
+        let resIds;
+        let displayNotif = false;
+        const state = this.model.get(this.handle, {raw: true});
+        if (this.isDomainSelected) {
+            resIds = await this._domainToResIds(state.getDomain(), session.active_ids_limit);
+            displayNotif = (resIds.length === session.active_ids_limit);
+        } else {
+            resIds = this.model.localIdsToResIds(this.selectedRecords);
+        }
+        return {resIds, displayNotif, total: this.isDomainSelected ? state.count : resIds.length};
```

Comment:
I would prefer this method to simply return a list of ids, and let the code in _toggleArchiveState deal with the displayNotif and total stuff